### PR TITLE
initial removal of z_zone.h macros

### PR DESCRIPTION
**Pull request by rfomin**
Done using the method suggested by @fabiangreffrath. The question is where to use zone memory and where stdlib. I suggest:
- `p_*`, `r_*` files only zone memory
- `i_*`, `m_*`, `deh.c`, `u_*` only stdlib
- `d_*`, `w_*` stdlib for all strings manipulation

### CODE REVIEW

**Comment by rfomin** [comment]
Not sure if I should fix this cppcheck error: https://stackoverflow.com/a/27589881

**Comment by rfomin** [comment]
> I guess once this is finished we can go without the parentheses around the stdlib function calls? Because, there is nothing left to unalias then, right?

Yes, of course, that is the whole point of these efforts. I have left parentheses to show where I prefer to use stdlib instead of zone memory.

> * `p_*`, `r_*` files only zone memory
> * `i_*`, `m_*`, `deh.c`, `u_*` only stdlib
> * `d_*`, `w_*` stdlib for all strings manipulation

What do you think of this approach?

**Comment by fabiangreffrath** [comment]
> Yes, of course, that is the whole point of these efforts. I have left parentheses to show where I prefer to use stdlib instead of zone memory.

Phew, alright, I was a bit confused by their still presence. 😉 

> What do you think of this approach?

Very good! Everything directly related to game logic and rendering (this could be expanded by e.g. `am_*`, `hu_*` and `st_*`) should use zone memory, whereas all "helper functions" should use stdlib

File: Source/z_zone.h [hunk 1]
```diff
@@ -80,20 +80,6 @@ void Z_DumpHistory(char *);
 #define Z_Realloc(a,b,c,d) (Z_Realloc)  (a,b,c,d,__FILE__,__LINE__)
 #define Z_CheckHeap()      (Z_CheckHeap)(__FILE__,__LINE__)
 
-// Remove all definitions before including system definitions
```
Comment (fabiangreffrath) [inline Source/z_zone.h:83]: Can we also kill lines 37-43?

File: Source/d_main.c [hunk 2]
```diff
@@ -598,8 +598,9 @@ void D_AddFile(const char *file)
   static int numwadfiles, numwadfiles_alloc;
 
   if (numwadfiles >= numwadfiles_alloc)
-    wadfiles = realloc(wadfiles, (numwadfiles_alloc = numwadfiles_alloc ?
-                                  numwadfiles_alloc * 2 : 8)*sizeof*wadfiles);
+    wadfiles = Z_Realloc(wadfiles, (numwadfiles_alloc = numwadfiles_alloc ?
+                                  numwadfiles_alloc * 2 : 8)*sizeof*wadfiles,
+                        PU_STATIC, 0);
```
Comment (fabiangreffrath) [inline Source/d_main.c:603]: Is this properly indented? It looks weird on 
https://github.com/fabiangreffrath/woof/pull/551/files

File: Source/am_map.c [hunk 3]
```diff
@@ -409,9 +409,10 @@ void AM_addMark(void)
   // remove limit on automap marks
 
   if (markpointnum >= markpointnum_max)
-    markpoints = realloc(markpoints,
+    markpoints = Z_Realloc(markpoints,
                         (markpointnum_max = markpointnum_max ? 
-                         markpointnum_max*2 : 16) * sizeof(*markpoints));
+                         markpointnum_max*2 : 16) * sizeof(*markpoints),
```
Comment (fabiangreffrath) [inline Source/am_map.c:414]: Is this properly indented?

File: Source/d_main.c (hunk 2 above)
Comment (rfomin) [reply Source/d_main.c:603]: Not sure how to fix it. BTW maybe we should use `I_Realloc` here?

**Comment by fabiangreffrath** [comment]
DO you consider this ready for merging?

**Comment by rfomin** [comment]
> Do you consider this ready for merging?

Yes, it's ready.

**Comment by fabiangreffrath** [comment]
Thank you very much, awesome work!

**Pull request merged**